src/sidebar/config/build-settings.js

```diff
@@ -178,55 +130,78 @@ async function fetchGroupsAsync(config, rpcCall) {
 }
 
 /**
- * Fetch the host configuration and merge it with the app configuration from h.
+ * Find and retrieve the correct host configuration for the sidebar.
  *
- * There are 3 ways to get the host config:
- *  Direct embed - From the hash string of the embedder frame.
- *  Legacy RPC with unknown parent - From a ancestor parent frame that passes it down via RPC. (deprecated)
- *  RPC with known parent - From a ancestor parent frame that passes it down via RPC.
+ * Host configuration may come from one of three places:
+ * - From the embedding (host) document frame.
+ *   The annotator encodes JavaScript/JSON settings from the host frame as a
+ *   URL fragment on the sidebar frame's `src`. In this (common) case,
+ *   use the parsed `configFromURL` itself as the host configuration.
+ * - From a known parent frame, requested by RPC (e.g. LMS).
+ *   Send a single RPC request to a known frame to retrieve host configuration
+ * - From an ancestor frame, using RPC (deprecated)
+ *   Send RPC requests for configuration to ancestor frames until we
+ *   (hopefully) find a host configuration.
  *
- * @param {SidebarConfig} appConfig
- * @param {Window} window_ - Test seam.
- * @return {Promise<MergedConfig>} - The merged settings.
+ * In cases where configuration is retrieved by RPC, the `configFromURL` object
+ * values are not retained in the returned host configuration. Note the
+ * exception related to the `group` property below.
+ *
+ * @param {HostConfig} configFromURL
+ * @param {Window} [window_]
+ * @returns {Promise<HostConfig>}
  */
-export async function fetchConfig(appConfig, window_ = window) {
-  const hostConfig = hostPageConfig(window);
-
-  const requestConfigFromFrame = hostConfig.requestConfigFromFrame;
+async function findHostConfiguration(configFromURL, window_) {
+  const requestConfigFromFrame = configFromURL.requestConfigFromFrame;
 
   if (!requestConfigFromFrame) {
-    // Directly embed: just get the config.
-    return fetchConfigEmbed(appConfig, hostConfig);
+    // In this case, the `configFromURL` is retained and used as host configuration
+    return configFromURL;
   }
+
   if (typeof requestConfigFromFrame === 'string') {
-    // Legacy: send RPC to all parents to find config. (deprecated)
-    // nb. Browsers may display errors in the console when messages are sent to frames
-    // that don't match the origin filter".
-    return await fetchConfigLegacy(appConfig, window_);
+    return await fetchConfigFromAncestorFrame(requestConfigFromFrame, window_);
```

Comment:
I think it may be wise to deploy a change to remove the legacy configuration approach first, and then do this refactoring. This will save needing to test/review changes to unused code.